src/main/java/com/patchworkmc/event/EventSubscriptionChecker.java

```diff
@@ -0,0 +1,70 @@
+package com.patchworkmc.event;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+// Currently patchwork cannot handle @SubscribeEvent for overloaded methods
+public class EventSubscriptionChecker {
+	private static class Entry {
+		public List<SubscribeEvent> subscribeEvents;
+		public List<String> superClasses;
+
+		Entry(List<SubscribeEvent> subscribeEvents, List<String> superClasses) {
+			this.subscribeEvents = subscribeEvents;
+			this.superClasses = superClasses;
+		}
+	}
+
+	private Map<String, Entry> data = new HashMap<>();
```

Comment:
`data` is not super descriptive, something like `entries` would be more informative.
```suggestion
	private Map<String, Entry> entries = new HashMap<>();
```